### ISSUE TITLE
Add function to predict travel times

### ISSUE THREAD

**Issue by kimjoaoun** [ropensci/stplanr#350]
I'd be great having a function that predicts travel times based on origin-destination data. I think it's possible if the OD dataset contains the travel time of each individual.

**Comment by Robinlovelace** [comment]
Numerous routing services, including ones with R interfaces such as those provided by [opentripplanner](https://github.com/ITSLeeds/opentripplanner) and [osrm](https://github.com/rCarto/osrm), give you time estimates. If you get an API key, as documented in `?route_cyclestreets()`, you can in fact already get travel times by cycle for many cities worldwide, as shown in the example below:

``` r
library(stplanr)
#> Registered S3 method overwritten by 'R.oo':
#>   method        from       
#>   throw.default R.methodsS3
nchar(Sys.getenv("CYCLESTREET"))
#> [1] 16
l = od2line(flow = od_data_sample, zones = cents_sf)
r = line2route(l[2:3, ], route_fun = stplanr::route_cyclestreet, plan = "quietest")
r[1:5]
#> Simple feature collection with 2 features and 5 fields
#> geometry type:  LINESTRING
#> dimension:      XY
#> bbox:           xmin: -1.550806 ymin: 53.82442 xmax: -1.516734 ymax: 53.82887
#> epsg (SRID):    4326
#> proj4string:    +proj=longlat +datum=WGS84 +no_defs
#>   plan start finish length time                       geometry
#> 1    1     1      1   1481  472 LINESTRING (-1.51667 53.828...
#> 2    1     1      1   3134  864 LINESTRING (-1.51667 53.828...
```

<sup>Created on 2019-10-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

If you want to do it all 'in R' it should be possible to generate time estimates with `route_local()` by dividing distances from `sf::st_length()` by average speed. **dodgr** @mpadge, the backend of `route_dodgr()` may be able to provide more accurate travel time estimates, Mark may have follow-up comments.

I'm planning to improve support for different routing services, particularly opentripplanner (which can estimate multi-modal and public transport trip times) and osrm. Would that solve your issue? 

If you could a reproducible example, with example data (e.g. a representive set of ~10 OD pairs), of a problem you want to solve, that will help understand exactly what you want to do. Any context you can provide will also help.

In summary, thanks for raising this important issue, lots of options, a reprex will help, and hope the info + links in this reply helps.
